### DISCUSSION TITLE
Detecting layout changes in tabs that are not the current one

### DISCUSSION THREAD

**Discussion by asmodeus812** [neovim/neovim#30124] [Q&A]
I am looking at being able to do what the title says, would like to update the tcd for the tab where the layout change has occured, the tcd depends on a lua function being run to compute and set it. I believe none of the available events would actually be useful, but maybe i am wrong. Doing that in the tabline string generating function is very much nogo, as the tabline plugin has no relation to computing/setting the tcd

**Comment by justinmk** [comment]
> layout changes in tabs that are not the current one

To change a tab's layout requires temporarily switching to that tab. Can you (minimal) example code that shows the scenario you are referring to?

**Reply by asmodeus812** [reply]
So here is a simple problem i have tcd set for each tab based on the currently active window and buffer for each tab. However, say a window in a non current tab page stops showing a buffer, for whatever reason, due to the buffer maybe being deleted/goes hidden, window gets closed, focused window in that other tabpage is set to show another buffer, or due to some other action. Then the focused window will change, and so would the active buffer (assume there are more than one window showing different buffers) for that non current tabpage, i would like to intercept that so i could set the the tcd.

**Reply by justinmk** [reply]
> say a window in a non current tab page stops showing a buffer, for whatever reason

Ok, I wouldn't call that a "layout" change. 

Did you try listening to the `BufWinLeave` event?

(This is why sample code saves time, it answers those kinds of questions.)
